versions/models.py

```diff
@@ -784,6 +784,61 @@ def __get__(self, instance, instance_type=None):
         else:
             return current_elt.__class__.objects.current.get(identity=current_elt.identity)
 
+    def get_prefetch_queryset(self, instances, queryset=None):
+        """
+        Overrides the parent method to:
+        - force queryset to use the querytime of the parent objects
+        - ensure that the join is done on identity, not id
+        - make the cache key identity, not id.
+        """
+        if queryset is None:
+            queryset = self.get_queryset()
+        queryset._add_hints(instance=instances[0])
+
+        # CleanerVersion change 1: force the querytime to be the same as the prefetched-for instance.
+        # This is necessary to have reliable results and avoid extra queries for cache misses when
+        # accessing the child objects from their parents (e.g. choice.poll).
+        instance_querytime = instances[0]._querytime
+        if instance_querytime.active:
+            if queryset.querytime.active and queryset.querytime.time != instance_querytime.time:
+                raise ValueError("A Prefetch queryset that specifies an as_of time must match "
+                                 "the as_of of the base queryset.")
+            else:
+                queryset.querytime = instance_querytime
+
+        # CleanerVersion change 2: make rel_obj_attr return a tuple with the object's identity.
+        # rel_obj_attr = self.field.get_foreign_related_value
+        def versioned_fk_rel_obj_attr(versioned_rel_obj):
+            return versioned_rel_obj.identity,
+        rel_obj_attr = versioned_fk_rel_obj_attr
+        instance_attr = self.field.get_local_related_value
+        instances_dict = {instance_attr(inst): inst for inst in instances}
+        # CleanerVersion change 3: fake the related field so that it provides a name of 'identity'.
+        # related_field = self.field.foreign_related_fields[0]
+        related_field = namedtuple('VersionedRelatedFieldTuple', 'name')('identity')
+
+        # FIXME: This will need to be revisited when we introduce support for
```

Comment:
Should we take over this comment or rather make a reference to the relative place in the Django base: https://github.com/django/django/blob/1.8/django/db/models/fields/related.py#L560 ?
Since, basically, this has not much to do with CleanerVersion itself, since "just" monkey-patching.